### ISSUE TITLE
Can't be able to use `generateHtmlFromNodes` correctly.

### ISSUE THREAD

**Issue by eashish93** [facebook/lexical#3616]
When using `generateHtmlFromNodes`, I encountered an error: 
`To use $generateHtmlFromNodes in headless mode please initialize a headless browser implementation such as JSDom before calling this function.` 
which is fine as JSDOM is required, but this error doesn't mention which variables this function needs, so I dig into the code and found that we need to declare something like this first to make it work:
```js
const { JSDOM } = jsdom;

const window = new JSDOM(`...`).window;
global.document = window.document;
// @ts-expect-error
global.window = window;

// Sometimes `global.navigator` also needs to be defined with `global.window `

// @ts-expect-error
global.DocumentFragment = () => JSDOM.fragment(``);
```
This function is checking for both `document` and `window`. This should only check for `document` and `DocumentFragment` only, as these are the only methods `generateHtmlFromNodes` using.

Lexical version: 0.7.4

**Comment by Multiply** [comment]
Same issue here. Overriding document, window etc. also seems to trip up NextJS's API runtime, so an option to pass in JSDOM as an override would be amazing.

**Comment by sneko** [comment]
> Same issue here. Overriding document, window etc. also seems to trip up NextJS's API runtime, so an option to pass in JSDOM as an override would be amazing.

@thegreatercurve @zurfyx any plan to add this? Would avoid polluting the backend environment as mentioned in https://github.com/facebook/lexical/issues/3097#issuecomment-1511496254

**Comment by Bunkerbewohner** [comment]
I also just stumbled over this. For context: I'm persisting the editor's collaboration state in form of the Ydoc using Hocuspocus and a custom persistence extension. On the server-side I'm storing/loading the Ydoc, and in addition I want to generate HTML from the Ydoc / editorstate so that I can server-side render an HTML preview of the editor before it's fully loaded.

**Comment by Fritiof** [comment]
I can't even get the JSDOM solution to work, happy to get an example to look at if anyone can share!

**Comment by Chigozie-Gerald** [comment]
> I also just stumbled over this. For context: I'm persisting the editor's collaboration state in form of the Ydoc using Hocuspocus and a custom persistence extension. On the server-side I'm storing/loading the Ydoc, and in addition I want to generate HTML from the Ydoc / editorstate so that I can server-side render an HTML preview of the editor before it's fully loaded.

You can create a webpack library in a nodejs workspace and have the document and window properties defined globally there or utilize webpack's `webpack.ProvidePlugin` plugin
Wrote a solution about it [here](https://github.com/facebook/lexical/issues/2020#issuecomment-2016945911)

**Comment by konhi** [comment]
[This implementation ](https://github.com/2wheeh/lexical-nextjs-ssr/blob/6d97581def739cb5763b5a3a2d611f8873a3013e/src/lib/index.ts)solves the issue for Next

**Comment by koderyoda** [comment]
> [This implementation ](https://github.com/2wheeh/lexical-nextjs-ssr/blob/6d97581def739cb5763b5a3a2d611f8873a3013e/src/lib/index.ts)solves the issue for Next

But other concurrent requests might fail while this function is running because this is still polluting global??

**Comment by etrepum** [comment]
No, it’s all synchronous, there’s no danger of polluting other requests assuming your finalizer is correct.

**Comment by etrepum** [comment]
No. This function runs synchronously, no other code can execute while it is running. JavaScript doesn’t have any parallelism.
